ddtrace/settings/config.py

```diff
@@ -607,16 +627,72 @@ def __setattr__(self, key, value):
         if key == "_config":
             return super(self.__class__, self).__setattr__(key, value)
         elif key in self._config:
-            self._config[key].set_code(value)
-            self._notify_subscribers([key])
+            self._set_config_items([(key, value, "code")])
             return None
         else:
             return super(self.__class__, self).__setattr__(key, value)
 
+    def _set_config_items(self, items):
+        # type: (List[Tuple[str, Any, _ConfigSource]]) -> None
+        for key, value, origin in items:
+            self._config[key].set_value_source(value, origin)
+
+        from ..internal.telemetry import telemetry_writer
+
+        cs = [{"name": k, "value": self._config[k].value(), "origin": self._get_source(k)} for k, _, _ in items]
+        telemetry_writer.add_event({"configuration": cs}, "app-client-configuration-change")
+        self._notify_subscribers([i[0] for i in items])
+
     def _reset(self):
         # type: () -> None
         self._config = _default_config()
 
     def _get_source(self, item):
         # type: (str) -> str
         return self._config[item].source()
+
+    def _remoteconfigPubSub(self):
+        from ddtrace.internal.remoteconfig._connectors import PublisherSubscriberConnector
+        from ddtrace.internal.remoteconfig._publishers import RemoteConfigPublisher
+        from ddtrace.internal.remoteconfig._pubsub import PubSub
+        from ddtrace.internal.remoteconfig._pubsub import RemoteConfigSubscriber
+
+        class _GlobalConfigPubSub(PubSub):
+            __publisher_class__ = RemoteConfigPublisher
+            __subscriber_class__ = RemoteConfigSubscriber
+            __shared_data__ = PublisherSubscriberConnector()
+
+            def __init__(self, callback):
+                self._publisher = self.__publisher_class__(self.__shared_data__, None)
+                self._subscriber = self.__subscriber_class__(self.__shared_data__, callback, "GlobalConfig")
+
+        return _GlobalConfigPubSub
```

Comment:
This class relies heavily on RC internals. Should we define `_GlobalConfigPubSub` in `ddtrace.internal.remoteconfig`?

Reply:
I believe this is the expected usage of these components? I don't see another way to initialize a remote config subscriber.

Reply:
@avara1986 is this the expected way to add a basic RC pubsub?

Reply:
yes it is! this is the way :wink: